src/ALE/MOM_regridding.F90

```diff
@@ -461,31 +506,220 @@ subroutine initialize_regridding(CS, GV, US, max_depth, param_file, mdl, coord_m
   elseif (index(trim(string),'HYBRID:')==1) then
-    ke = GV%ke; allocate(dz(ke))
-    ! The following assumes the FILE: syntax of above but without "FILE:" in the string
+    ke = GV%ke
+    allocate(dz(ke))
     allocate(rho_target(ke+1))
+    ! The following assumes the FILE: syntax of above but without "FILE:" in the string
+    varName = trim( extractWord(trim(string(8:)), 3) )
+    if (varname == " ") call MOM_error(FATAL, &
+      trim(mdl)//", initialize_regridding: HYBRID "// &
+      "Too few arguments in ("//trim(string)//")")
     fileName = trim( extractWord(trim(string(8:)), 1) )
     if (fileName(1:1)/='.' .and. filename(1:1)/='/') fileName = trim(inputdir) // trim( fileName )
-    if (.not. file_exists(fileName)) call MOM_error(FATAL,trim(mdl)//", initialize_regridding: HYBRID "// &
+    if (.not. file_exists(fileName)) call MOM_error(FATAL, &
+      trim(mdl)//", initialize_regridding: HYBRID "// &
       "Specified file not found: Looking for '"//trim(fileName)//"' ("//trim(string)//")")
     varName = trim( extractWord(trim(string(8:)), 2) )
-    if (.not. field_exists(fileName,varName)) call MOM_error(FATAL,trim(mdl)//", initialize_regridding: HYBRID "// &
+    if (.not. field_exists(fileName,varName)) call MOM_error(FATAL, &
+      trim(mdl)//", initialize_regridding: HYBRID "// &
       "Specified field not found: Looking for '"//trim(varName)//"' ("//trim(string)//")")
     call MOM_read_data(trim(fileName), trim(varName), rho_target)
     varName = trim( extractWord(trim(string(8:)), 3) )
     if (varName(1:5) == 'FNC1:') then ! Use FNC1 to calculate dz
       call dz_function1( trim(string((index(trim(string),'FNC1:')+5):)), dz )
     else ! Read dz from file
-      if (.not. field_exists(fileName,varName)) call MOM_error(FATAL,trim(mdl)//", initialize_regridding: HYBRID "// &
+      if (.not. field_exists(fileName,varName)) call MOM_error(FATAL, &
+        trim(mdl)//", initialize_regridding: HYBRID "// &
         "Specified field not found: Looking for '"//trim(varName)//"' ("//trim(string)//")")
       call MOM_read_data(trim(fileName), trim(varName), dz)
     endif
     if (main_parameters) then
       call log_param(param_file, mdl, "!"//coord_res_param, dz, &
                trim(message), units=coordinateUnits(coord_mode))
       call log_param(param_file, mdl, "!TARGET_DENSITIES", rho_target, &
-               'HYBRID target densities for interfaces', units=coordinateUnits(coord_mode))
+               'HYBRID target densities for interfaces', units="kg m-3")
     endif
+  elseif (index(trim(string),'HYBRID_3D:')==1) then
+    ke = GV%ke
+    allocate(dz_3d(SZI_(G),SZJ_(G),ke), source=0.0)
+    allocate(rho_target_3d(SZI_(G),SZJ_(G),ke+1), source=0.0)
+    ! The following assumes the FILE: syntax of above but without "FILE:" in the string
+    varName = trim( extractWord(trim(string(11:)), 3) )
+    if (varname == " ") call MOM_error(FATAL, &
+      trim(mdl)//", initialize_regridding: HYBRID_3D "// &
```

Comment:
This is a very minor quibble with an otherwise exceptional contribution, but the MOM6 code style guide (https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide) suggests an indentation of at least 4 spaces for continuation lines.  (I do recognize that there are lots of other lines already in this file that do not follow this style convention.)

Reply:
@awallcraft, if you object strongly to adding the indentation to follow the style guide in the new parts of this file, I would not have a problem if you were to just resolve this conversation without actually making the changes.
  However, that would then run the risk that a lot of these lines would get changed in a later PR if we ever figured out how to automatically implement that part of the style guide recommendations, or if someone were feeling like they were in a picky mood about enforcing the style guide!